src/Text2Image/T2IParamInput.cs

```diff
@@ -463,6 +469,32 @@ static T2IParamInput()
         PromptTagLengthEstimators["embed"] = PromptTagLengthEstimators["preset"];
         PromptTagLengthEstimators["embedding"] = PromptTagLengthEstimators["preset"];
         PromptTagLengthEstimators["lora"] = PromptTagLengthEstimators["preset"];
+
+        PromptTagProcessors["setvar"] = (data, context) =>
+        {
+            string varname = context.PreData ?? "";
+            data = context.Parse(data);            
+            if (varname != "")
+            {
+                context.RandomVariables[varname] = data ?? "";
+            }
+            return data;
+        };
+        PromptTagLengthEstimators["setvar"] = (data) =>
+        {
+            return ProcessPromptLikeForLength(data);  //Should really be 0, since the setvar doesn't add anything to the prompt, but it accounts for one use of the variable
+        };
+
+        PromptTagProcessors["var"] = (data, context) =>
+        {           
+            if (data == null) { return ""; }
+            return (context.RandomVariables.TryGetValue(data, out string value) ? (value ?? "") : "");
```

Comment:
should probably have a warning if var name is unrecognized rather than just blank